modules/cloud-function/README.md

```diff
@@ -0,0 +1,160 @@
+# Cloud Function Module
+
+Cloud Function management, with support for IAM roles and optional bucket creation.
+
+The GCS object used for deployment uses a hash of the bundle zip contents in its name, which ensures change tracking and avoids recreating the function if the GCS object is deleted and needs recreating.
+
+## TODO
+
+- [ ] add support for source repository
+
+## Examples
+
+### HTTP trigger
+
+This deploys a Cloud Function with an HTTP endpoint, using a pre-existing GCS bucket for deployment, setting the service account to the Cloud Function default one, and delegating access control to the containing project.
+
+```hcl
+module "cf-http" {
+  source        = "../modules/net-cloudnat"
+  project_id    = "my-project"
+  name          = "test-cf-http"
+  bucket_name   = "test-cf-bundles"
+  bundle_config = {
+    source_dir = "my-cf-source-folder
+    output_path = "bundle.zip"
+  }
+}
+```
+
+### Non-HTTP triggers
+
+Other trigger types other than HTTP are configured via the `trigger_config` variable. This example shows a PubSub trigger.
+
+```hcl
+module "cf-http" {
+  source        = "../modules/net-cloudnat"
+  project_id    = "my-project"
+  name          = "test-cf-http"
+  bucket_name   = "test-cf-bundles"
+  bundle_config = {
+    source_dir = "my-cf-source-folder
+    output_path = "bundle.zip"
+  }
+  trigger_config = {
+    event = "google.pubsub.topic.publish"
+    resource = local.my-topic
+    retry = null
+  }
+}
+```
+
+### Anonymous HTTP access
```

Comment:
nit: I see above that you suggest handling IAM on the container project. BTW, since we handle in the module the Anonymous scenario, would you add an example to enable it for a specific user/service account?